File: basic math/c++/all_divisors.cpp
#include <iostream>
using namespace std;

void all_divisors(int n)
{
    for (int i = 1; i * i <= n; i++)
    {
        if (n % i == 0)
        {
            cout << i << " ";
            if (i != n / i)
                cout << n / i << " ";
            cout << endl;
        }
    }
}

int main()
{
    int n;
    cin >> n;
    all_divisors(n);
    return 0;
}

File: basic math/javascript/all_divisors.js
const readline = require('readline')

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
})

function allDivisors(number) {
    let divisors = []
    for (let i = 1; i*i <= number; i++) {
        if (number % i === 0) {
            divisors.push(i)
            if (i !== number/i) {
                divisors.push(number/i)
            }
        }
    }
    return divisors
}

rl.question('Enter a number: ', (number) => {
    number = parseInt(number)
    console.log(allDivisors(number))
    rl.close()
})